gordo/machine/dataset/base.py

```diff
@@ -115,6 +119,14 @@ def join_timeseries(
 
         """
         resampled_series = []
+        series_iterable = list(series_iterable)
```

Comment:
Carefull! This loads all the series in memory, and they can be big. The rest of the stuff is intended to be lazy (using generator).

Reply:
ya ya.. trying to think of a way around this. :(